### PR TITLE
Bob error handling

### PR DESCRIPTION
**Pull request by da-kami**

### CODE REVIEW

File: swap/src/bin/swap_cli.rs [hunk 1]
```diff
@@ -118,8 +118,16 @@ async fn main() -> Result<()> {
             );
             let (swap, event_loop) = bob_factory.with_init_params(send_bitcoin).build().await?;
 
-            tokio::spawn(async move { event_loop.run().await });
-            bob::run(swap).await?;
+            let handle = tokio::spawn(async move { event_loop.run().await });
+            let swap = bob::run(swap);
+            tokio::select! {
+                event_loop_result = handle => {
+                    event_loop_result??;
```
Comment (thomaseizinger) [inline swap/src/bin/swap_cli.rs:125]: I wonder if Rust will ever add an exclamation mark operator: `??!!?!`

File: swap/src/protocol/bob/event_loop.rs [hunk 2]
```diff
@@ -193,7 +193,7 @@ impl EventLoop {
                         }
                         OutEvent::ResponseSent => {}
                         OutEvent::CommunicationError(err) => {
-                            error!("Communication error: {:#}", err)
+                            anyhow!("Communication error: {:#}", err);
```
Comment (thomaseizinger) [inline swap/src/protocol/bob/event_loop.rs:196]: This doesn't actually terminate the loop, you are just constructing an error. You need to either explicitly say `return Err(anyhow!(""))` or use `bail!` which implies the `return`.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]

File: swap/src/protocol/bob/event_loop.rs (hunk 2 above)
Comment (da-kami) [reply swap/src/protocol/bob/event_loop.rs]: Fixed - changed to `bail!`! For some reason I thought that `anyhow` is equivalent to `bail`, but as you pointed out it does not return. How would it ever return whit a questionmark in the end. What a noob mistake 🤦‍♂️

File: swap/src/bin/swap_cli.rs (hunk 1 above)
Comment (da-kami) [reply swap/src/bin/swap_cli.rs:125]: Eineinself!

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**